src/Functions/makeDate.cpp

```diff
@@ -96,56 +65,101 @@ class FunctionMakeDate : public FunctionWithNumericParamsBase
 
     String getName() const override { return name; }
 
-    bool isVariadic() const override { return false; }
-
-    size_t getNumberOfArguments() const override { return argument_names.size(); }
-
     DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
     {
-        checkRequiredArguments(arguments, argument_names, 0);
+        if (arguments.size() != argument_names_month_day.size() && arguments.size() != argument_names_dayofyear.size())
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Function {} requires {} or {} arguments, but {} given",
+                getName(), argument_names_month_day.size(), argument_names_dayofyear.size(), arguments.size());
+
+        for (size_t i = 0; i < arguments.size(); ++i)
+        {
+            DataTypePtr argument_type = arguments[i].type;
+            if (!isNumber(argument_type))
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                        "Argument '{}' for function {} must be a number",
+                        (arguments.size() == argument_names_month_day.size()) ? argument_names_month_day[i] : argument_names_dayofyear[i], getName());
+
+        }
 
         return std::make_shared<typename Traits::ReturnDataType>();
     }
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
         Columns converted_arguments;
-        convertRequiredArguments(arguments, argument_names, converted_arguments);
+        const DataTypePtr converted_argument_type = std::make_shared<DataTypeFloat32>();
+        converted_arguments.clear();
```

Comment:
Do we need to clear it?

Reply:
`reserve()` probably also got lost. Why did we need to move it from the base class?

Reply:
Thanks for  checking! (I wanted to ask @davenger since he is the original author of `makeDate()`)

True, there is no need to clear `converted_arguments`, that was only a leftover.

About `reserve()`: In terms of performance, it doesn't really matter if it is there or not, vector `converted_arguments` has only two or three elements, and the loop runs only once per chunk (as initialization). I guess not having it is a little less clutter.

I rewrote the code to use the more high-level `FunctionArgumentDescriptors`/`validateFunctionArgumentTypes`. That should make it easier to grasp what the code is doing.